fiona/_crs.pyx

```diff
@@ -20,8 +20,38 @@ logger = logging.getLogger(__name__)
 cdef int OAMS_TRADITIONAL_GIS_ORDER = 0
 
 
+cdef _osr_to_wkt(OGRSpatialReferenceH cogr_srs, crs, wkt_version):
+    cdef char *wkt_c = NULL
+    wkt = None
+    cdef const char* options_wkt[2]
+    options_wkt[0] = NULL
+    options_wkt[1] = NULL
+
+    if wkt_version:
+        wkt_format = f"FORMAT={wkt_version}".encode("utf-8")
+        options_wkt[0] = wkt_format
+    OSRExportToWktEx(cogr_srs, &wkt_c, options_wkt)
+    if wkt_c != NULL:
+        wkt_b = wkt_c
+        wkt = wkt_b.decode('utf-8')
+
+    if not wkt and wkt_version is None:
+        # attempt to morph to ESRI before export
+        options_wkt[0] = "FORMAT=WKT1_ESRI"
+        OSRExportToWktEx(cogr_srs, &wkt_c, options_wkt)
+
+    if wkt_c != NULL:
+        wkt_b = wkt_c
+        wkt = wkt_b.decode('utf-8')
+
+    _cpl.CPLFree(wkt_c)
+
+    if not wkt:
+        raise CRSError("Invalid input to create CRS: {}".format(crs))
```

Comment:
This error should somehow also describe `wkt_version`, e.g. `fiona._crs.crs_to_wkt(target_crs, wkt_version="abc")` will raise an error that makes no mention of a potentially bad `wkt_version` value.

Not sure if it's possible to capture possible invalid `wkt_version` values earlier, or if that's a bad idea, as other values work, e.g. `wkt_version="WKT2"`.

Reply:
there are a few ways to capture an error message, like this https://github.com/Toblerity/Fiona/blob/9c8fe736d4381526905c6976c399a33d10e3ecbf/fiona/ogrext.pyx#L1512-L1516

Reply:
Thanks!

Current error message:
```
fiona.errors.CRSError: Invalid input to create CRS: EPSG:4326. Please choose a different WKT version (WKT2 is recommended): WKT1 | WKT1_GDAL | WKT1_ESRI | WKT2 | WKT2_2018 | WKT2_2015. GDAL ERROR: Unsupported value for FORMAT
```

Reply:
Useful error capture, as sometimes `WKT1_GDAL` will be specified, but the error message would contain (e.g.) "GDAL ERROR: PROJ: proj_as_wkt: Unsupported conversion method: Lambert Conic Conformal (2SP Michigan)" as the chosen wkt format is incompatible.